components/dashboards-web-component/src/viewer/components/WidgetReportConfigurationDialog.jsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { Component } from 'react';
+import PropTypes from "prop-types";
+import { Checkbox, FlatButton, Dialog } from 'material-ui';
+import DashboardReportGenerator from '../../utils/DashboardReportGenerator';
+
+export default class WidgetReportConfigurationDialog extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            dialogOpen: false,
+            includeTime: false,
+        };
+
+        this.handleClose = this.handleClose.bind(this);
+        this.handleOpen = this.handleOpen.bind(this);
+        this.handleIncludeGenerateTime = this.handleIncludeGenerateTime.bind(this);
+        this.handleIncludeNoOfRecords = this.handleIncludeNoOfRecords.bind(this);
+        this.generateWidgetReport = this.generateWidgetReport.bind(this);
+    }
+
+    generateWidgetReport() {
+        this.handleClose();
+        const element = this.props.widget;
+        const docTitle = this.props.title;
+        DashboardReportGenerator.generateWidgetPdf(element, docTitle, this.state.includeTime, this.state.includeRecords,
+            this.props.themeName);
+        this.state.includeRecords = false;
```

Comment:
Do not write to state directly! refer: https://reactjs.org/docs/react-component.html#state

What is the reason to do this?

Reply:
Done